lib/eventFactory.js

```diff
@@ -686,17 +792,29 @@ class EventFactory extends BaseFactory {
                                     .then(creator => {
                                         modelConfig.creator = creator;
 
+                                        let adaptedScmUri = pipeline.scmUri;
+
+                                        if (subscribedEvent) {
+                                            adaptedScmUri = subscribedScmConfig.scmUri;
+                                        }
+
                                         return this.scm.decorateCommit({
-                                            scmUri: pipeline.scmUri,
+                                            scmUri: adaptedScmUri,
                                             scmContext,
                                             sha,
                                             token
                                         });
                                     });
                             }
 
+                            let adaptedScmUri = pipeline.scmUri;
+
+                            if (subscribedEvent) {
+                                adaptedScmUri = subscribedScmConfig.scmUri;
+                            }
+
                             return this.scm.decorateCommit({
-                                scmUri: pipeline.scmUri,
+                                scmUri: adaptedScmUri,
```

Comment:
why is this change needed ? decoration is still against current pipeline's scm commit right ?